### PR TITLE
[STRAT-114] Routes file finalized

### PR DESCRIPTION
**Pull request by ufshk**
Routes file completed, still working on tests, finalizing elevations with Krishna, finalizing speed limits as well.

PR for the heck of it. Only one file; rest will come soon. Elevs nearly done (in fact I think it is done but I have to review it and let Krishna know). Tests to be fixed up soon after. 

We will need to talk about the Speed Limits file cause I realized today that it works but is sort of very unoptimized. I have removed the class structure from it, didn't seem to need anything else... until I checked it out further today. It does return data as of now, so that's not the problem. Discuss sometime this week?

We aren't tracking these on the JIRA board which is why the ticket is STRAT-???.

### CODE REVIEW

File: routemodel/data_retrieval/routes.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+         
+    return params
+
+def get_routing_data(points: str, route_attrs = 'routePath', dist_unit = 'km'):
+    """
+    Getting navigation data response from Bing Maps API.
+    @param points: string of points formatted for requesting from Bing Maps API.
+        Use points_query_builder method to build string correctly.
+    @param dist_unit (optional): Either 'km' or 'mi,' default to km since
+        we're Canadian, eh.
```
Comment (e-wai) [inline routemodel/data_retrieval/routes.py:44]: Excellent

File: routemodel/data_retrieval/routes.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    @param route_attrs (optional):
```
Comment (e-wai) [inline routemodel/data_retrieval/routes.py:45]: Can we list the options? 6 doesn't seem outlandish

File: routemodel/data_retrieval/routes.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            # parse through the items in each route leg
+            for item in route_leg['itineraryItems']:
+                    # relevant info parsed below
+                    lat = item['maneuverPoint']['coordinates'][0]
+                    longit = item['maneuverPoint']['coordinates'][1]
+                    instruction = item['instruction']['text']
+                    dist = item['travelDistance']
+                    direction = item['compassDirection']
+
+                    # ensure street name is found
```
Comment (e-wai) [inline routemodel/data_retrieval/routes.py:87]: Sorry, just for clarification- what are the two options here? Why is street name sometimes first element, sometimes econd?

**Review by e-wai — DISMISSED** [review]
I've yet to compare with Krishna's stuff, but everything seems OK to me. Mostly comments for clarification

**Comment by e-wai** [comment]
Just quick note - think Krishna should merge first; off the top of my head he deletes `config_example` and you add to it so that might be a fun rebasing problem to keep in mind?

**Comment by mdshiozaki** [comment]
Strat 115 in JIRA

**Comment by ufshk** [comment]
> 
> 
> Strat 115 in JIRA

It seems to be 114?

**Review by e-wai — APPROVED** [review]

File: routemodel/data_retrieval/routes.py (hunk 3 above)
Comment (ufshk) [reply routemodel/data_retrieval/routes.py:95]: to be more clear, sometimes the street name is in the first element, sometimes in the second element from the API response.

**Review by e-wai — APPROVED** [review]

**Review by mdshiozaki — APPROVED** [review]

**Pull request merged**